### ISSUE TITLE
(CI) Prepopulate Nix cache for Linux runners is too slow

### ISSUE THREAD

**Issue by steve-chavez** [PostgREST/postgrest#3131]
## Problem

On my two latest PRs ([ref1](https://github.com/PostgREST/postgrest/pull/3129), [ref2](https://github.com/PostgREST/postgrest/pull/3130)), the "Prepopulate Nix cache for Linux runners" job took too much time:

- 17 minutes: https://github.com/PostgREST/postgrest/actions/runs/7268604462/job/19804764119?pr=3129
- 23 minutes: https://github.com/PostgREST/postgrest/actions/runs/7277759274/job/19830490472?pr=3130

This is bad because all the "Test PG X.Y" jobs depend on it. It makes development much slower.

cc @develop7 @laurenceisla

**Comment by steve-chavez** [comment]
The upside seems to be that the next prepopulate job runs fast:

- 1 min: https://github.com/PostgREST/postgrest/actions/runs/7278227283/job/19831963552?pr=3130

Only the first run is slow it seems.

**Comment by develop7** [comment]
I'd suggest extracting the prepopulating job to a separate workflow and schedule it to run once or twice a day against the "main" branch. This way PRs will pick generated caches up too.

**Comment by laurenceisla** [comment]
Yes, I also noticed the increased time in my PRs. 

> I'd suggest extracting the prepopulating job to a separate workflow and schedule it to run once or twice a day against the "main" branch. 

That's a good alternative, since caches cannot be shared between sibling branches, only with the base one (`main` in this case). Then, perhaps the other jobs needn't be dependent on the Prepopulate one and just use the cache directly. 

I still think the Prepopulate job takes too much time, since it builds PostgREST at least twice among other things. Maybe only the relevant tools (used only by the CI jobs) should be installed?

**Comment by develop7** [comment]
> I still think the Prepopulate job takes too much time, since it builds PostgREST at least twice among other things. 

Fair enough. We could reuse postgrest outputs further down the pipeline, but if the build fails, it'd  

> Maybe only the relevant tools (used only by the CI jobs) should be installed?

That was my initial intention, but with the list I've managed to collect it didn't improve really. So I've just instantiated everything and let it go, since the effort was already taking too much.

**Comment by develop7** [comment]
@steve-chavez these runs' logs you've mentioned do not seem to be out of order to me, more looks like the network congestion issue. Scheduled cache warmups should make us less dependent on it.
